### PR TITLE
fix: throw new SdkHttpClient error from default HTTP Client

### PR DESCRIPTION
**Pull request by WojtekTheWebDev**
### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSDoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [ ] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📝 Checklist

- [ ] I have linked an issue or discussion.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts [hunk 1]
```diff
@@ -491,4 +495,72 @@ describe("middlewareModule", () => {
...
+            });
+
+            return data;
+          } catch (err: any) {
+            throw new SdkHttpError({
+              statusCode: err?.response?.status || 500,
+              message: err?.response?.data?.message || err.message,
+              cause: err,
+            });
+          }
```
Comment (michaelKurowski) [inline packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts:545]: I've thought you wanted to return the error like I've shown you. Did you change your mind or is it an accident?

**Review by jagoral — APPROVED** [review]

File: packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts [hunk 2]
```diff
@@ -491,4 +495,72 @@ describe("middlewareModule", () => {
...
+        isSpecificSdkHttpError(err, {
+          statusCode: (statusCode) => statusCode === 401,
+        })
+      ).toBe(true);
+      expect(isSdkRequestError(err)).toBe(true);
+      expect(isSdkUnauthorizedError(err)).toBe(true);
+    }
+  });
+
+  it("should throw an error as an instance of SdkHttpError class with axios as a custom http client", async () => {
```
Comment (michaelKurowski) [inline packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts:524]: What you're really testing here is 
"should make a method throw if `httpClient` callback throws"
Neither axios nor the error is part of the SDK, instead it's part of the test's code.

File: packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts [hunk 3]
```diff
@@ -491,4 +495,72 @@ describe("middlewareModule", () => {
...
+    expect.assertions(6);
+
+    const sdk = initSDK({
+      commerce: buildModule(middlewareModule<Endpoints>, {
+        apiUrl: "http://localhost:8181/commerce",
+        httpClient: async (url, params, config) => {
```
Comment (michaelKurowski) [inline packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts:530]: I'm also thinking, it looks like it's `fetch`, not a client. Client suggests that it's an instance of something, instead, it's a fetcher.

**Review by michaelKurowski — APPROVED** [review]
I've posted few things that may be helpful, but they're minor and I won't block the PR because of them. I trust your judgement in what to do with them.

File: packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts (hunk 1 above)
Comment (WojtekTheWebDev) [reply packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts:545]: I abandoned this idea as the response of fetch, axios etc are almost always any and this trick does not work :(

File: packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts (hunk 2 above)
Comment (WojtekTheWebDev) [reply packages/sdk/src/__tests__/integration/modules/middlewareModule.spec.ts]: Applied.

**Pull request merged**